base/io.jl

```diff
@@ -669,8 +669,34 @@ function readuntil_indexable(io::IO, target#=::Indexable{T}=#, out)
     if done(target, first)
         return true
     end
-    len = endof(target)
+
+    # In readuntil, we use a cache to accelerate scanning of sequences
```

Comment:
I think it would be better to give the algorithm's name (KMP), since we didn't invent this. https://en.wikipedia.org/wiki/Knuth–Morris–Pratt_algorithm

Also, doesn't feel entirely accurate, since the first entry of the cache ("second") is actually unrolled